### PR TITLE
Dev/update badge overflow and table

### PR DESCRIPTION
**Pull request by kirill-kruchkov**
## Purpose

- Update [OverflowMenu](/#/Components?id=overflowmenu) looks.
- Default [OverflowMenu](/#/Components?id=overflowmenu) appearance is now `secondary`.
- Update [Badge](/#/Components?id=badge) to have `radius`, `size` and `inverse` props. Don't rely on parent font-size anymore.
- Update sizes in all components, use small font size for controls.
- Increase [Table.Cell](/#/Components?id=table) vertical padding.
- Use fashionTrade eslint config.

### Visualization

Screenshot 2019-04-10 at 13 00 33
Screenshot 2019-04-10 at 13 00 40
Screenshot 2019-04-10 at 13 00 47
Screenshot 2019-04-10 at 13 00 55
Screenshot 2019-04-10 at 13 02 05
Screenshot 2019-04-10 at 13 02 14

### Checklist

- [x] Update changelog

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -1,28 +1,5 @@
...
-    "react/forbid-prop-types": ['error', { forbid: ['any'] }]
-  },
-  "globals": {
-    "document": true,
-    "window": true
-  },
-  "env": {
-    "jest/globals": true
-  }
+  extends: '@fashiontrade/eslint-config-ft-react-app',
```
Comment (zoheiry) [inline .eslintrc.js]: Nice 😉

File: src/components/Button/Button.js [hunk 2]
```diff
@@ -142,14 +140,14 @@ const Button = styled(ButtonInner)`
...
 
 Button.propTypes = {
   /** Button label */
   children: PropTypes.node.isRequired,
   className: PropTypes.string,
-  size: PropTypes.oneOf(['small', 'normal']),
+  size: PropTypes.oneOf(['normal', 'large']),
   disabled: PropTypes.bool,
   type: PropTypes.oneOf(['button', 'reset', 'submit']),
   appearance: PropTypes.oneOf(['primary', 'secondary', 'link']),
```
Comment (zoheiry) [inline src/components/Button/Button.js]: Do you think you can also change this to support "success", "warning", etc.. appearances. If it's a lot of work then we can do that in a separate PR.

File: src/components/OverflowMenu/OverflowMenu.js [hunk 3]
```diff
@@ -3,17 +3,21 @@ import PropTypes from 'prop-types';
...
 import noop from '../../helpers/noop';
+import Icon from '../Icon';
 import ToggleMenu from '../ToggleMenu';
 import DropdownItem from '../DropdownItem';
 import DropdownTitle from '../DropdownTitle';
 
 const Trigger = styled.div`
   display: flex;
-  padding: 8px;
+  padding: calc(0.25rem - 1px);
```
Comment (zoheiry) [inline src/components/OverflowMenu/OverflowMenu.js]: Why the -1px?

File: src/components/OverflowMenu/OverflowMenu.js [hunk 4]
```diff
@@ -3,17 +3,21 @@ import PropTypes from 'prop-types';
...
   text-align: center;
   cursor: pointer;
-  transition: 0.2s linear;
-  border: solid 1px ${p => (p.active ? p.color : 'transparent')};
+  transition: border-color 0.1s linear, color 0.1s linear;
+  color: ${p => p.color};
+  border: solid 1px ${p => p.color};
```
Comment (zoheiry) [inline src/components/OverflowMenu/OverflowMenu.js]: I saw some designs without the border, and if we always have a border then the showroom will content blocks will look a bit strange, should we have the border or no border configurable `hideBorder: PropTypes.bool`?

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,5 +1,19 @@
 ## [Unreleased]
 
+### Added
+
+- `map-marker`, `pencil`, `shop`, and `truck` icons.
+
+### Changed
+
+- Update [OverflowMenu](/#/Components?id=overflowmenu) looks.
```
Comment (zoheiry) [inline CHANGELOG.md]: I'd consider this a breaking change since adding a border might drastically change how it's displayed in some places.

File: src/components/OverflowMenu/OverflowMenu.js [hunk 6]
```diff
@@ -62,6 +66,21 @@ class OverflowMenu extends Component {
...
+  };
+
+  getActiveColor = () => {
+    const { appearance, theme } = this.props;
+    const colors = {
+      primary: theme.primaryActive,
+      secondary: theme.primary,
+      success: theme.success,
+      warning: theme.warning,
+      danger: theme.danger,
```
Comment (zoheiry) [inline src/components/OverflowMenu/OverflowMenu.js]: Shouldn't the active color be different than the default color? either that or we add an outline when active.

File: src/theme/sizes.js [hunk 7]
```diff
@@ -5,12 +5,12 @@ export const GRID_GUTTER_WIDTH = 32;
...
 // Control paddings
 export const PADDING_HORIZONTAL_LARGE = 32;
-export const PADDING_HORIZONTAL_BASE = 16;
+export const PADDING_HORIZONTAL_BASE = 24;
 export const PADDING_HORIZONTAL_SMALL = 8;
 
 export const PADDING_VERTICAL_SMALL = 6;
-export const PADDING_VERTICAL_BASE = 8;
-export const PADDING_VERTICAL_LARGE = 12;
+export const PADDING_VERTICAL_BASE = 10;
```
Comment (dinkra) [inline src/theme/sizes.js]: Not 8 friendly? 😱

File: src/theme/sizes.js [hunk 8]
```diff
@@ -23,13 +23,13 @@ const sizes = {
...
+  paddingHorizontalSmall: `${PADDING_HORIZONTAL_SMALL / FONT_SIZE_SMALL}rem`,
+  paddingHorizontalBase: `${PADDING_HORIZONTAL_BASE / FONT_SIZE_BASE}rem`,
+  paddingHorizontalLarge: `${PADDING_HORIZONTAL_LARGE / FONT_SIZE_LARGE}rem`,
 
-  paddingVerticalSmall: `${PADDING_VERTICAL_SMALL / FONT_SIZE_SMALL}em`,
-  paddingVerticalBase: `${PADDING_VERTICAL_BASE / FONT_SIZE_BASE}em`,
-  paddingVerticalLarge: `${PADDING_VERTICAL_LARGE / FONT_SIZE_LARGE}em`,
+  paddingVerticalSmall: `${PADDING_VERTICAL_SMALL / FONT_SIZE_SMALL}rem`,
+  paddingVerticalBase: `${PADDING_VERTICAL_BASE / FONT_SIZE_BASE}rem`,
+  paddingVerticalLarge: `${PADDING_VERTICAL_LARGE / FONT_SIZE_LARGE}rem`,
```
Comment (zoheiry) [inline src/theme/sizes.js]: Getting ready for mobile I see 😄

File: src/components/Searchbar/Searchbar.js [hunk 9]
```diff
@@ -14,21 +15,9 @@ const Wrapper = styled('div')`
   color: inherit;
 `;
 
-const Input = styled('input')`
+const Input = styled(BaseInput)`
```
Comment (zoheiry) [inline src/components/Searchbar/Searchbar.js]: ❤️

File: src/theme/sizes.js (hunk 7 above)
Comment (kirill-kruchkov) [reply src/theme/sizes.js]: So the thing is these are real internals. They are build to match with used font-size + line-height so that the whole control (e.g. `Button`) matches the grid, though not `8px` grid but `4px` grid (that's how our controls are now).

Large button has normal `line-height=20px` plus `12*2` => `44px` (I'll question Daniela on that, this is new).

Normal button is `line-height=16px` + `2 *10` => `36px` (so Button's height hasn't changed, smaller font but bigger paddings).

 I know that's confusing. I'm sorry about that. That's why I really want to build this `Box` component, and stop exporting these.

File: src/components/Button/Button.js (hunk 2 above)
Comment (kirill-kruchkov) [reply src/components/Button/Button.js]: Oh ya, I can do that for sure!

File: src/components/OverflowMenu/OverflowMenu.js (hunk 4 above)
Comment (kirill-kruchkov) [reply src/components/OverflowMenu/OverflowMenu.js]: I'll ask Daniela. Maybe we should indeed.

File: src/components/OverflowMenu/OverflowMenu.js (hunk 3 above)
Comment (kirill-kruchkov) [reply src/components/OverflowMenu/OverflowMenu.js]: So this is a 8px-grid-aligned padding and minus one px to compensate for the border

File: CHANGELOG.md (hunk 5 above)
Comment (kirill-kruchkov) [reply CHANGELOG.md]: Yeah, some of these changes should indeed be considered breaking. I'll change it here.

File: src/theme/sizes.js (hunk 8 above)
Comment (kirill-kruchkov) [reply src/theme/sizes.js]: I tried to keep also the dependency on current font-size you introduced, but it becomes super complicated then to maintain it thinking about several use cases (sizes via props, current font-size, how they align with a grid, etc.) so I decided to try this approach. I didn't met any usages with current `font-size` yet in `app-service`. Maybe we have a couple, but I guess we should change them then to align with the designs.
